Lib/distutils/dist.py

```diff
@@ -26,6 +26,18 @@
 # to look for a Python module named after the command.
 command_re = re.compile(r'^[a-zA-Z]([a-zA-Z0-9_]*)$')
 
+def _ensure_list(value, fieldname):
+    if isinstance(value, str):
+        # a string containing comma separated values is okay.  It will
+        # be converted to a list by Distribution.finalize_options().
+        pass
+    elif not isinstance(value, list):
+        # passing a tuple or an iterator perhaps, warn and convert
+        typename = type(value).__name__
+        msg = f"Warning: '{fieldname}' should be a list, got type '{typename}'"
```

Comment:
Is the prefix `Warning` redundant?  Otherwise message seems good.

Reply:
Another style nit: Can't we use `{fieldname!r}` instead of `'{fieldname}'`?